server/src/main/java/org/elasticsearch/search/aggregations/support/AggregationContext.java

```diff
@@ -230,6 +230,12 @@ public final AggregationUsageService getUsageService() {
      */
     public abstract Analyzer getIndexAnalyzer(Function<String, NamedAnalyzer> unindexedFieldAnalyzer);
 
+    /**
+     * Is this request cacheable? Requests that have
+     * non-deterministic queries or scripts aren't cachable.
+     */
+    public abstract boolean isCacheable();
```

Comment:
These are required by tests. I could avoid adding it with some terrible casting stuff in tests but I think it is better to be up front about it.